### PR TITLE
string permissions

### PR DESCRIPTION
**Pull request by EliasEriksson**
* interfaces used in bodies of requests/responses now have string permissions.
  * This change cascades a lot. many isly types and tests changed due to this change.
* removed users property from `Organization.Creatable` users are now added via `Organization.Changeable`
* modified functions under `User.Permissions` to take both string and object versions
* added a few more functions under `User.Permissions` to make it easier to manipulate the permission strings
* bumped minor version

### CODE REVIEW

File: User/Permissions/index.ts [hunk 1]
```diff
@@ -43,63 +44,122 @@ export namespace Permissions {
...
-						permissions[organization] == true ||
-						permissions["*"] == true
+						parsed[organization] == true ||
+						parsed["*"] == true
 				)
 		}
 		return result
 	}
+	export function get(permissions: string, organization: Organization.Identifier | "*"): string
+	export function get<T extends Permissions<flagly.Flags>>(
```
Comment (EliasEriksson) [inline User/Permissions/index.ts:62]: add some tests

File: User/Permissions/index.ts [hunk 2]
```diff
@@ -43,63 +44,122 @@ export namespace Permissions {
...
 		)
-		return result as Permissions
+		return typeof target == "object" ? (result as T) : flagly.Flags.stringify(result)
+	}
+	export function filter<T extends Permissions<flagly.Flags>>(
+		permissions: T,
+		organizations: Organization.Identifier[]
+	): T
+	export function filter(permissions: string, organizations: Organization.Identifier[]): string
+	export function filter<T extends Permissions<flagly.Flags>>(
```
Comment (EliasEriksson) [inline User/Permissions/index.ts:146]: add some tests

File: User/Permissions/index.ts [hunk 3]
```diff
@@ -43,63 +44,122 @@ export namespace Permissions {
...
+	): string | T {
+		const parsed = typeof permissions == "object" ? permissions : flagly.parse(permissions)
+		const result = Object.fromEntries(Object.entries(parsed).filter(([id]) => organizations.includes(id))) as T
+		return typeof permissions == "object" ? result : flagly.Flags.stringify(result)
 	}
-	// Readable only for backwards compatibility
-	export type Readable<T extends flagly.Flags = flagly.Flags> = Permissions<T>
-	export namespace Readable {
-		export const is = type.is
+	export function organizations(
```
Comment (EliasEriksson) [inline User/Permissions/index.ts:158]: add some tests

**Review by EliasEriksson — COMMENTED** [review]
add tests to new functions

**Review by linusarnejohansson — CHANGES_REQUESTED** [review]

File: User/Key/Creatable.spec.ts [hunk 4]
```diff
@@ -9,43 +9,21 @@ describe("Creatable", () => {
...
-		const type = userwidgets.User.Key.Creatable.type.create({ claims, permissions })
+		type Creatable = userwidgets.User.Key.Creatable<Claims>
+		const type = userwidgets.User.Key.Creatable.type.create({ claims })
 		const creatable: Creatable = {
 			id: "r0ck3t",
 			name: { first: "jessie", last: "doe" },
 			email: "jessie@rocket.com",
-			permissions: { "o--o1--o": { user: { view: true }, foo: false } },
-		}
-		const dirty = {
```
[the remark this answers is not in the document]
Comment (EliasEriksson) [reply User/Key/Creatable.spec.ts:39]: kind of. its not testable. there are no interfaces for strings isly can clean up. its a string.

**Review by linusarnejohansson — APPROVED** [review]

**Pull request merged**